packages/utils/src/lib/scoring.ts

```diff
@@ -0,0 +1,122 @@
+import {
+  AuditGroup,
+  AuditReport,
+  CategoryConfig,
+  CategoryConfigRefType,
+  PluginReport,
+  Report,
+} from '@code-pushup/models';
+
+type EnrichedAuditReport = AuditReport & { plugin: string };
+type EnrichedScoredAuditGroup = AuditGroup & { plugin: string; score: number };
+type ScoredCategoryConfig = CategoryConfig & { score: number };
+
+export type ScoredReport = Omit<Report, 'plugins' | 'categories'> & {
+  plugins: (Omit<PluginReport, 'audits' | 'groups'> & {
+    audits: EnrichedAuditReport[];
+    groups: EnrichedScoredAuditGroup[];
+  })[];
+  categories: ScoredCategoryConfig[];
+};
```

Comment:
The `ScoredReport` maybe needs to get iterated multiple times in a single report or in multiple formats. This would help the lookup time.

I thought about a complete refactor to a nested category dict but it did for some reason not show up in the perf benchmark... I have to create a proper setup there in a next PR to get TS support and be able to easily import parts of the code like in a test.

Reply:
I like the dictionary idea, that would be most elegant.

Reply:
I already have a issue open